lib/src/transformer/declaration_parsing.dart

```diff
@@ -56,8 +56,18 @@ class ParsedDeclarations {
     unit.declarations.forEach((CompilationUnitMember member) {
       member.metadata.forEach((annotation) {
         var name = annotation.name.toString();
+        var publicMember;
 
-        declarationMap[name]?.add(member);
+        if (member is ClassDeclaration && member.name.name.startsWith('_\$')) {
+          unit.declarations.forEach((CompilationUnitMember innerLoopMember) {
+            // Look for a matching public class.
+            if (innerLoopMember is ClassDeclaration && member.name.name.substring(2) == innerLoopMember.name.name) {
+              publicMember = innerLoopMember;
+            }
+          });
+        }
+
+        declarationMap[name]?.add(publicMember ?? member);
```

Comment:
Might be a good idea to create either a separate key in this map for the public props members, or add it to a different data structure entirely. This would allow the removing of the `with` clause to be more DRY and more robust, since we can just iterate over those class declarations directly instead of picking them out when iterating over props, abstract props, state, and abstract state in `impl_generation.dart`.